lib/manageiq/api/common/user.rb

```diff
@@ -0,0 +1,79 @@
+module ManageIQ
+  module API
+    module Common
+      class User
+        IDENTITY_KEY = 'x-rh-identity'.freeze
+
+        def self.current
+          Thread.current[:attr_current_user]
+        end
+
+        def self.current=(user_header)
+          Thread.current[:attr_current_user] = new(user_header)
+        end
+
+        def self.with_logged_in_user(user_header)
+          alternate_user = new(user_header)
```

Comment:
@syncrou Shouldn't this use the self.current= and set the value? as opposed to using alternate user.

This call will typically get used by the Background Queue process e.g. When Provisioning gets an approval from Approval Service Message Bus it would fetch the saved headers from the DB and call this function to set the User.current to mimic as if it came from the Controller.

Reply:
@mkanoor - I don't have any issue using `self.current=`, though it seemed there was no use for the `Thread.current` (TLS) when used inside a block so I decided to use a local var instead.  Would it be better to set `Thread.current` and ensure it is nilled out like my original code?

Reply:
@syncrou We are going to be writing methods that will use
the User.current_user for example when sending out Webhook payload or calling Approval Service with the name of the user that created the request. So we don't have a choice we we have to set the TLS so that the underlying methods be consistent when they are called from the Rails Controller or a background job engine like Sidekiq where there is no logged in user but we have access to the previously stored headers in the Order.

Reply:
@mkanoor - I'll get that change in.